ProxyAgent/Proxy.cs

```diff
@@ -198,7 +199,14 @@ private IHttpRequest BuildRequest(HttpRequest requestIn, string toHostname)
             var method = requestIn.Method.ToUpperInvariant();
             if (HttpClient.HttpClient.MethodsWithPayload.Contains(method))
             {
-                requestOut.SetContent(this.GetRequestPayload(requestIn));
+                if (requestIn.ContentType.ToLowerInvariant().StartsWith(IMAGE))
```

Comment:
You might actually consider to always use binary transfer, that would work right? If I remember correctly, that's what the service does with the response.

The approach here looks a little bit hacky, I would suggest to find a list of common binary mediatypes, make it a list, and check for the value (also use case insensitive comparison). e.g.
* `image/png`
* `image/jpeg`
* `image/gif`
* `application/octet-stream`
* `application/zip`
* `application/pdf`
* `application/msword`
etc.